backend/views/views.py

```diff
@@ -34,12 +38,44 @@ def admin():
     return render_template('admin_page.html')
 
 
-@app.route('/userpage')
+@app.route('/userpage', methods=['GET', 'POST'])
 @admin_permissions
 def user_page():
-    users = db.session.query(User, Role).filter(
-        User.role_id == Role.id).order_by(User.id).all()
-    return render_template('user_page.html', users=users)
+    form = SearchForm(request.args, csrf_enabled=False)
+    if  form.validate():
+        key = int(request.args.get('field_by'))
+        search_string = str(request.args.get('search'))
+        condition_list = []
+        for one_string in search_string.split():
+            if len(one_string) < MIN_SEARCH_STR:
+                continue
+            search_parameter = '%{}%'.format(one_string)
+            name_search = User.name.like(search_parameter)
+            alias_search = User.alias.like(search_parameter)
+            email_search = User.email.like(search_parameter)
+            conditions = [
+                name_search,
+                alias_search,
+                email_search,
+                or_(name_search, alias_search),
+                or_(alias_search, email_search),
+                or_(email_search, name_search),
+                or_(name_search, alias_search, email_search)
+                ]
+            condition_list.append(conditions[key])
+        condition = or_(condition_list[x] for x in range(len(condition_list)))
```

Comment:
So try `condition = or_(*condition_list)`. Anyway I have no idea why your expression works :)

Reply:
I experimented with it for a long time:) It is work